packages/universal-components/src/RangeDatePicker/DraggableItem.ios.js

```diff
@@ -0,0 +1,101 @@
+// @flow
+
+import React from 'react';
+import { Animated } from 'react-native';
+import { PanGestureHandler, State } from 'react-native-gesture-handler';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+
+import { designTokens } from '../DesignTokens';
+import { StyleSheet, type StylePropType } from '../PlatformStyleSheet';
+import type { GrabbedSideType } from './RangeDatePickerTypes';
+
+// @TODO flow types for events
+
+type Props = {
+  +style?: StylePropType,
+  +onDrag: (any, GrabbedSideType) => void,
+  +onDrop: any => void,
+  +grabbedSide: GrabbedSideType,
+};
+
+type XY = {
+  x: number,
+  y: number,
+};
+
+export default class DraggableItem extends React.Component<Props> {
+  constructor(props: Props) {
+    super(props);
+    this.translateX = new Animated.Value(0);
+    this.translateY = new Animated.Value(0);
+    this.lastOffset = { x: 0, y: 0 };
+    this.onGestureEvent = Animated.event(
+      [
+        {
+          nativeEvent: {
+            translationX: this.translateX,
+            translationY: this.translateY,
+          },
+        },
+      ],
+      {
+        useNativeDriver: true,
+        listener: this.onDrag,
+      },
+    );
+  }
+
+  lastOffset: XY;
+  translateY: Animated.Value;
+  translateX: Animated.Value;
+  onGestureEvent: any;
+
+  onDrag = (event: any) => {
```

Comment:
Could we maybe define some event types, maybe it could be in `universal-components` 🤔 
something like [this](https://github.com/kiwicom/mobile/blob/master/packages/shared/types/Events.js)

Reply:
I was also thinking about this solution.  But the event is quite big object:
```
{
  "_dispatchInstances": FiberNode {
    "tag": 7,
    "key": null,
    "type": "RCTView",
  },
  "_dispatchListeners": [Function bound value],
  "_targetInst": FiberNode {
    "tag": 7,
    "key": null,
    "type": "RCTView",
  },
  "bubbles": undefined,
  "cancelable": undefined,
  "currentTarget": 933,
  "defaultPrevented": undefined,
  "dispatchConfig": Object {
    "registrationName": "onGestureHandlerEvent",
  },
  "eventPhase": undefined,
  "isDefaultPrevented": [Function J],
  "isPropagationStopped": [Function J],
  "isTrusted": undefined,
  "nativeEvent": Object {
    "absoluteX": 294,
    "absoluteY": 285.5,
    "handlerTag": 2,
    "numberOfPointers": 1,
    "state": 4,
    "target": 933,
    "translationX": 111.5,
    "translationY": -12.5,
    "velocityX": 53.02986135148045,
    "velocityY": 0,
    "x": 18.5,
    "y": 17.5,
  },
  "target": 933,
  "timeStamp": 1556178367135,
  "type": undefined,
}
```

Would it be enough type just `nativeEvent` without exact type using `|`?

Reply:
Just add the values we are using, we can expand it as needed.

Reply:
OK, done